app/src/ui/rebase/rebase-branch-dialog.tsx

```diff
@@ -0,0 +1,189 @@
+    this.setState({ filterText })
+  }
+
+  private onSelectionChanged = (selectedBranch: Branch | null) => {
+    this.setState({ selectedBranch })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  public render() {
+    const selectedBranch = this.state.selectedBranch
+    const currentBranch = this.props.currentBranch
+
+    const selectedBranchIsNotCurrentBranch =
+      selectedBranch === null ||
+      currentBranch === null ||
+      currentBranch.name === selectedBranch.name
+
+    const disabled = selectedBranchIsNotCurrentBranch
+
+    // the amount of characters to allow before we truncate was chosen arbitrarily
+    const currentBranchName = truncateWithEllipsis(
+      this.props.currentBranch.name,
+      40
+    )
+
+    const loading = this.state.isRebasing ? <Loading /> : null
+
+    return (
+      <Dialog
+        id="rebase"
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.startRebase}
+      >
+        <DialogHeader
+          title={
+            <div className="rebase-dialog-header">
+              Rebase <strong>{currentBranchName}</strong> onto…
+            </div>
+          }
+          dismissable={true}
+          onDismissed={this.props.onDismissed}
+        />
+        <DialogContent>
+          <BranchList
+            allBranches={this.props.allBranches}
+            currentBranch={currentBranch}
+            defaultBranch={this.props.defaultBranch}
+            recentBranches={this.props.recentBranches}
+            filterText={this.state.filterText}
+            onFilterTextChanged={this.onFilterTextChanged}
+            selectedBranch={selectedBranch}
+            onSelectionChanged={this.onSelectionChanged}
+            canCreateNewBranch={false}
+            renderBranch={this.renderBranch}
+          />
+        </DialogContent>
+        <DialogFooter>
+          <ButtonGroup>
+            <Button type="submit" disabled={disabled}>
+              {loading}
+              Rebase <strong>
+                {currentBranch ? currentBranch.name : ''}
+              </strong>{' '}
+              onto <strong>{selectedBranch ? selectedBranch.name : ''}</strong>
+            </Button>
+          </ButtonGroup>
+        </DialogFooter>
+      </Dialog>
+    )
+  }
+
+  private startRebase = async () => {
+    const branch = this.state.selectedBranch
+    if (!branch) {
+      return
+    }
+
+    this.setState({ isRebasing: true })
+
+    await this.props.dispatcher.rebase(
+      this.props.repository,
+      branch.name,
+      this.props.currentBranch.name
+    )
+
+    this.setState({ isRebasing: false })
+  }
+
+  /**
+   * Returns the branch to use as the selected branch
+   *
+   * The initial branch is used if passed
+   * otherwise, the default branch will be used iff it's
+   * not the currently checked out branch
+   */
+  private resolveSelectedBranch() {
```

Comment:
Could we extract this out into a regular function that takes these three parameters rather than accessing the instance variable props? Not a huge fan of invoking instance methods from the constructor.

Reply:
Done in d3f2dbdd7a50e57dbcf9fe0aeb2c4af55dcc61ba